mmhuman3d/core/visualization/renderer/torch3d_renderer/meshes.py

```diff
@@ -0,0 +1,540 @@
+        super().__init__(
+            verts=verts,
+            faces=faces,
+            textures=textures,
+            verts_normals=verts_normals,
+        )
+
+    def get_faces_padded(self, N_batch, N_individual):
+        faces = self.face_individual.repeat(N_batch, N_individual, 1)
+        faces_offset = torch.arange(N_individual).view(N_individual, 1).repeat(
+            1, self.model_class.NUM_FACES).view(1, -1, 1).to(faces.device)
+        faces += faces_offset
+        return faces
+
+    def _compute_list(self):
+        self._faces_list = self.faces_list()
+        self._verts_list = self.verts_list()
+
+    def extend(self, N_batch: int, N_scene: int = 1):
+        if N_batch == 1:
+            meshes_batch = self
+        else:
+            meshes_batch = join_meshes_as_batch([self for _ in range(N_batch)])
+
+        if N_scene == 1:
+            meshes = meshes_batch
+        else:
+            meshes = join_batch_meshes_as_scene(
+                [meshes_batch for _ in range(N_scene)])
+        return meshes
+
+    def clone(self):
+        """Modified from pytorch3d and add `model_type` in
+        __class__.__init__."""
+        verts_list = self.verts_list()
+        faces_list = self.faces_list()
+        new_verts_list = [v.clone() for v in verts_list]
+        new_faces_list = [f.clone() for f in faces_list]
+        other = self.__class__(
+            verts=new_verts_list,
+            faces=new_faces_list,
+            model_type=self.model_type)
+        for k in self._INTERNAL_TENSORS:
+            v = getattr(self, k)
+            if torch.is_tensor(v):
+                setattr(other, k, v.clone())
+
+        # Textures is not a tensor but has a clone method
+        if self.textures is not None:
+            other.textures = self.textures.clone()
+        return other
+
+    def detach(self):
+        """Modified from pytorch3d and add `model_type` in
+        __class__.__init__."""
+        verts_list = self.verts_list()
+        faces_list = self.faces_list()
+        new_verts_list = [v.detach() for v in verts_list]
+        new_faces_list = [f.detach() for f in faces_list]
+        other = self.__class__(
+            verts=new_verts_list,
+            faces=new_faces_list,
+            model_type=self.model_type)
+
+        for k in self._INTERNAL_TENSORS:
+            v = getattr(self, k)
+            if torch.is_tensor(v):
+                setattr(other, k, v.detach())
+
+        # Textures is not a tensor but has a detach method
+        if self.textures is not None:
+            other.textures = self.textures.detach()
+        return other
+
+    def update_padded(self, new_verts_padded: torch.Tensor):
+        """Modified from pytorch3d and add `model_type` in
+        __class__.__init__."""
+
+        def check_shapes(x, size):
+            if x.shape[0] != size[0]:
+                raise ValueError(
+                    'new values must have the same batch dimension.')
```

Comment:
have the same batch size

Reply:
Copied from pytorch3d, but `batch size` is better